include/cadmium/simulation/rt_clock/mbedclock.hpp

```diff
@@ -0,0 +1,230 @@
+        	expired = false;
+          	long timeLeft = delay_us;
+
+		executionTimer.reset();
+
+        	// Handle waits of over ~35 minutes as timer overflows
+        	while (!interrupted && (timeLeft > INT_MAX)) {
+        	 	this->expired = false;
+        	 	this->_timeout.attach_us(callback(this, &MBEDClock::timeout_expired), INT_MAX);
+
+        	 	while (!expired && !interrupted) {
+				sleep();
+			}
+
+        	 	if(!interrupted){
+        	 		timeLeft -= INT_MAX;
+        	 	}
+        	}
+
+        	//Handle waits of under INT_MAX microseconds
+        	if(!interrupted && timeLeft > 0) {
+        		this->_timeout.attach_us(callback(this, &MBEDClock::timeout_expired), timeLeft); // sets callback to change expire after timeLeft time
+        		while (!expired && !interrupted) {
+				sleep(); // remains here until expired is set true by the _timeout callback
+			}
+        	}
+
+		// if timeLeft < 0 there is no wait
+
+        	executionTimer.stop();
+        	expired = false;
+
+		// in case the sleep() is interrupted by an external interrupt return the remaining time, otherwise return 0
+        	if(interrupted) {
+        	 	timeLeft -= executionTimer.read_us();
+        	 	if(delay_us < timeLeft ) {
+				return 0;
+			}
+        	 	//hal_critical_section_enter();
+        	 	// interrupted = false;
+        	 	return delay_us - timeLeft;
+        	}
+
+        	return 0;
+        }
+
+     protected:
+        T rTimeLast; //!< last real system time.
+        long  maxJitter; //!< Maximum allowed delay jitter. This parameter is optional.
+     public:
+        volatile bool interrupted; // for external interrupts
+
+        //! The empty constructor does not check the accumulated delay jitter.
+        MBEDClock(): RealTimeClock(), executionTimer(), _timeout() {
+		rTimeLast = 0.0;
+		maxJitter = -1;
+		expired = false;
+		interrupted = false;// used for interrupts
+	}
+
+	/**
+	 * Use this constructor to select the maximum allowed delay jitter.
+	 * @param maxJitter duration of the maximum allowed jitter.
+	 */
+	[[maybe_unused]] explicit MBEDClock(long max_jitter): MBEDClock() {
+		maxJitter = max_jitter;
+	}
+
+
+        /**
+         * Starts the real-time clock.
+         * @param timeLast initial simulation time.
+         */
+        void start(double timeLast) override {
+            RealTimeClock::start(timeLast);
+	    rTimeLast = 0.0;
+	    executionTimer.reset();
+            executionTimer.start();
+        }
+
+        /**
+         * Stops the real-time clock.
+         * @param timeLast last simulation time.
+         */
+        void stop(double timeLast) override {
+            // rTimeLast = executionTimer.read_us();
+	    executionTimer.stop();
+            RealTimeClock::stop(timeLast);
+        }
+
+        /**
+         * Waits until the next simulation time.
+         * @param nextTime next simulation time (in seconds).
+         */
+        void waitUntil(double timeNext) override {
```

Comment:
I guess this is the method we need to change, right? Instead of `void`, it must return the actual `time` (which, by the way, must be less than or equal to `timeNext`)